src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,311 @@
+
+    private static List<Tuple2<Long, URI>> rawFASTQFiles;
+    private static Map<Long, URI> expectedAssembledFASTAFiles;
+
+    // parameter block
+    private static final int lengthDiffTolerance = 1;   // tolerance on how much of a difference can each pair of assembled contigs (actual vs expected) length could be
+    private static final int editDistanceTolerance = 3; // tolerance of the summed edit distance between expected contigs and corresponding actual assembled contigs
+
+    // set up path and data
+    @BeforeClass
+    private static void setup(){
+        String sgaPathString = System.getProperty("path.to.sga");
+        if(null==sgaPathString){
+            sgaPathString = System.getenv("path.to.sga");
+        }
+        sgaPath =  (null==sgaPathString) ? null : Paths.get(sgaPathString);
+
+        rawFASTQFiles = new ArrayList<>();
+        expectedAssembledFASTAFiles = new HashMap<>();
+
+        Long i = 0L;
+
+        final String[] extension = new String[1];
+        extension[0] = "fastq";
+
+        final Iterator<File> it = FileUtils.iterateFiles(TEST_DATA_DIR, extension, false);
+
+        while(it.hasNext()){
+
+            final String fastqFileName = it.next().getAbsolutePath().toString();
+            final Tuple2<Long, URI> breakpoint = RunSGAViaProcessBuilderOnSpark.assignFASTQToBreakpoints(String.valueOf(i) + "\t" + Paths.get(fastqFileName).toUri());
+            rawFASTQFiles.add(breakpoint);
+
+            final String contigFileName = fastqFileName.replace("fastq", "pp.ec.filter.pass.rmdup.merged-contigs.fa");
+            expectedAssembledFASTAFiles.put(i++, Paths.get(contigFileName).toUri());
+        }
+    }
+
+    @BeforeMethod
+    public static void checkPATHTOSGAAvailability(){
+
+        if(null==sgaPath){
+            throw new SkipException("Skipping test because \"-Dpath.to.sga\" is set neither in system property nor as an environment variable.");
+        }
+    }
+
+    @Test(groups = "sv")
+    public void assemblyOneStepTest() throws IOException, InterruptedException, RuntimeException{
+
+        for(final Tuple2<Long, URI> breakpoint : rawFASTQFiles){
+
+            final FileSystem fs = FileSystem.getLocal(new Configuration());
+
+            final RunSGAViaProcessBuilderOnSpark.PipeLineResult result = RunSGAViaProcessBuilderOnSpark.performAssembly(fs, breakpoint, sgaPath, true)._2();
+
+            final File expectedAssembledContigFile = new File( expectedAssembledFASTAFiles.get(breakpoint._1()) );
+            final List<String> expectedFASTAContents = (null==expectedAssembledContigFile) ? null : Files.readAllLines(Paths.get(expectedAssembledContigFile.getAbsolutePath()));
+            final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedContigsCollection = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(expectedFASTAContents);
+
+            compareContigs(result.sgaStepsResult.assembledContigs, expectedContigsCollection);
```

Comment:
\[FindBugs\] INFO: NP: Read of unwritten public or protected field sgaStepsResult in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.assemblyOneStepTest\(\)